android_jni/avifandroidjni/src/main/java/com/avif/android/AvifDecoder.java

```diff
@@ -0,0 +1,58 @@
+package com.avif.android;
+
+import android.graphics.Bitmap;
+import java.nio.ByteBuffer;
+
+/** An AVIF Decoder. AVIF Specification: https://aomediacodec.github.io/av1-avif/. */
+@SuppressWarnings("CatchAndPrintStackTrace")
+public class AvifDecoder {
+  static {
+    try {
+      System.loadLibrary("avif_android");
+    } catch (UnsatisfiedLinkError exception) {
+      exception.printStackTrace();
+    }
+  }
+
+  // This is a utility class and cannot be instantiated.
+  private AvifDecoder() {}
+
+  /** Contains information about the AVIF Image dimensions. */
+  public static class Info {
+    public int width;
+    public int height;
+  }
+
+  /**
+   * Returns true if the bytes in the buffer seem like an AVIF image.
+   *
+   * @param buffer The encoded image.
+   * @return true if the bytes seem like an AVIF image, false otherwise.
+   */
+  public static boolean isAvifStream(ByteBuffer buffer) {
+    return AvifDecoder.isAvifStream(buffer, buffer.remaining());
```

Comment:
IMPORTANT: I think it is okay to use `buffer.remaining()` here only when `buffer.position()` is 0. It seems better to use `buffer.limit()` instead().

Reply:
buffer.position() has to be 0 for the native calls to works. The native code can only access a pointer to the buffer at position 0 (irrespective of what the buffer's current position is).

So all the calls in this class assume that buffer.position() will be 0. So it should be ok to use both limit() and remaining().

I have clarified this requirement in the comments of all the functions.